# 
# Submitted-By: Robert Schwebel, 2006-01-18
# Committed-By: Robert Schwebel
#
# Error: 
#
# ttymsg.c: In function 'ttymsg':
# ttymsg.c:135: error: invalid lvalue in assignment
# make[1]: *** [ttymsg.o] Error 1
# make[1]: Leaving directory
# `/home/rsc/svn/ptxdist-build-0.7-trunk-pii-native/build-target/inetutils-1.4.2/libinetutils'
# make: ***
# [/home/rsc/svn/ptxdist-build-0.7-trunk-pii-native/state/inetutils.compile]
# Error 2
#
# Description: 
#
# make gcc4 compilable 
#
# State:
# 
# Patch taken from linux from scratch
#
#
# Submitted By: Matthew Burgess <matthew@linuxfromscratch.org>
# Date: 2005-08-30
# Initial Package Version: 1.4.2
# Upstream Status: From upstream
# Origin: Greg Schafer <gschafer@zip.com.au> (http://www.diy-linux.org/downloads/patches/GCC4/inetutils-1.4.2-gcc4-fixes-1.patch)
# Description: Fix compilation issues with gcc4

diff -Naur inetutils-1.4.2.orig/ftp/ftp_var.h inetutils-1.4.2/ftp/ftp_var.h
--- inetutils-1.4.2.orig/ftp/ftp_var.h	2000-07-08 01:00:53.000000000 +0000
+++ inetutils-1.4.2/ftp/ftp_var.h	2005-08-30 20:25:27.205932304 +0000
@@ -36,8 +36,6 @@
 #include <sys/param.h>
 #include <setjmp.h>
 
-#include "extern.h"
-
 #ifndef FTP_EXTERN
 #define FTP_EXTERN extern
 #endif
@@ -129,3 +127,5 @@
 FTP_EXTERN int macnum;			/* number of defined macros */
 FTP_EXTERN struct macel macros[16];
 FTP_EXTERN char macbuf[4096];
+
+#include "extern.h"
diff -Naur inetutils-1.4.2.orig/libinetutils/ttymsg.c inetutils-1.4.2/libinetutils/ttymsg.c
--- inetutils-1.4.2.orig/libinetutils/ttymsg.c	2001-11-01 15:52:19.000000000 +0000
+++ inetutils-1.4.2/libinetutils/ttymsg.c	2005-08-30 20:24:27.168059448 +0000
@@ -132,7 +132,7 @@
 	    }
 	  if (wret)
 	    {
-	      (char *)iov->iov_base += wret;
+	      iov->iov_base = (char *)iov->iov_base + wret;
 	      iov->iov_len -= wret;
 	    }
 	  continue;
